File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        int start = 0 , end = x, ans = -1;
        if(x == 1)return 1;
        while(start <= end){
            long long int mid = start + (end - start)/2;
            long long int sq = mid * mid;
            if(sq == x)return mid;
            else if (sq < x){
                ans = mid;
                start = mid + 1 ;
            }else end = mid - 1 ;
        }
        return ans;
    }
};

File: 0069-sqrtx/0069-sqrtx.java
class Solution {
    public int mySqrt(int x) {
        int s = 0 , e = x;
        long  ans = -1;
        while( s <= e){
            long  m = s + ( e - s )/2;
            long   sq = m * m;
            if(sq <= x){
                ans = m;
                s = (int)m +1 ;
            }else e = (int)m-1;
        }
        return (int)ans;
    }
}